docs/strategy-callbacks.md

```diff
@@ -568,3 +569,110 @@ class AwesomeStrategy(IStrategy):
+
+
+class DigDeeperStrategy(IStrategy):
+    
+    position_adjustment_enable = True
+    
+    # Attempts to handle large drops with DCA. High stoploss is required.
+    stoploss = -0.30
+    
+    # ... populate_* methods
+    
+    # Example specific variables
+    max_dca_orders = 3
+    # This number is explained a bit further down
+    max_dca_multiplier = 5.5
+    
+    # This is called when placing the initial order (opening trade)
+    def custom_stake_amount(self, pair: str, current_time: datetime, current_rate: float,
+                            proposed_stake: float, min_stake: float, max_stake: float,
+                            **kwargs) -> float:
+        
+        # We need to leave most of the funds for possible further DCA orders
+        # This also applies to fixed stakes
+        return proposed_stake / self.max_dca_multiplier
+        
+    def adjust_trade_position(self, trade: Trade, current_time: datetime,
+                              current_rate: float, current_profit: float, min_stake: float,
+                              max_stake: float, **kwargs):
+        """
+        Custom trade adjustment logic, returning the stake amount that a trade should be increased.
+        This means extra buy orders with additional fees.
+ 
+        :param trade: trade object.
+        :param current_time: datetime object, containing the current datetime
+        :param current_rate: Current buy rate.
+        :param current_profit: Current profit (as ratio), calculated based on current_rate.
+        :param min_stake: Minimal stake size allowed by exchange.
+        :param max_stake: Balance available for trading.
+        :param **kwargs: Ensure to keep this here so updates to this won't break your strategy.
+        :return float: Stake amount to adjust your trade
+        """
+        
+        if current_profit > -0.05:
+            return None
+
+        # Obtain pair dataframe (just to show how to access it)
+        dataframe, _ = self.dp.get_analyzed_dataframe(trade.pair, self.timeframe)
+        # Only buy when not actively falling price.
+        last_candle = dataframe.iloc[-1].squeeze()
+        previous_candle = dataframe.iloc[-2].squeeze()
+        if last_candle['close'] < previous_candle['close']:
+            return None
+
+        filled_buys = trade.select_filled_orders('buy')
+        count_of_buys = len(filled_buys)
+        
+        # Allow up to 3 additional increasingly larger buys (4 in total)
+        # Initial buy is 1x
+        # If that falls to -5% profit, we buy 1.25x more, average profit should increase to roughly -2.2%
+        # If that falls down to -5% again, we buy 1.5x more
+        # If that falls once again down to -5%, we buy 1.75x more
+        # Total stake for this trade would be 1 + 1.25 + 1.5 + 1.75 = 5.5x of the initial allowed stake.
+        # That is why max_dca_multiplier is 5.5
+        # Hope you have a deep wallet!
+        if 0 < count_of_buys <= self.max_dca_orders:
```

Comment:
Great job to both of you! I have a question about the `select_filled_orders` method. 

Wouldn't this condition trigger multiple orders if the first order doesn't get filled right away? I think this count should include the potential pyramidal orders created in the last cycle. 

As far as I understand, we're counting just filled orders with [these statuses](https://github.com/freqtrade/freqtrade/blob/c12f2378dbaa9bff3d315c70e93e3362cf68fadf/freqtrade/constants.py#L57) which would not select orders we might have already placed at this level and which are still open.

Reply:
Please read the documentation that was added as part of this PR.

this callback will not be called if a order is open.

Reply:
Sorry I missed it! It might be worth adding it to the function docstring as well.